src/main/kotlin/me/lensvol/blackconnect/BlackdClient.kt

```diff
@@ -51,8 +52,10 @@ class BlackdClient(val hostname: String, val port: Int) {
         skipStringNormalization: Boolean = false,
         targetPythonVersions: String = ""
     ): Result<BlackdResponse, Exception> {
-        val url = URL("http://$hostname:$port")
-
+        var url = URL("http://$hostname:$port")
+        if (useSsl) {
+            url = URL("https://$hostname:$port")
+        }
```

Comment:
At the moment, there are two approaches in the code that basically do the same thing. 

I suggest we merge them into one using property:

```
val blackdUrl: String
    get() =  URL("$protocol://$hostname:$port")
```

Reply:
>     val blackdUrl: URL
        get() =  URL("$protocol://$hostname:$port")

you mean this as a property of BlackdClient

Reply:
Yes :)

Reply:
private or public? => seems like public could deliver extra functionality ... later on (and allows me to write some extra tests :-)

Reply:
I think private should be enough for now.

Reply:
oops moved to public :-S